src/Files.App/Filesystem/FilesystemOperations/ShellFilesystemOperations.cs

```diff
@@ -525,37 +471,17 @@ public async Task<IStorageHistory> MoveItemsAsync(IList<IStorageItemWithPath> so
             var moveResult = new ShellOperationResult();
             if (sourceRename.Any())
             {
-                var (status, response) = await connection.SendMessageForResponseAsync(new ValueSet()
-                {
-                    { "Arguments", "FileOperation" },
-                    { "fileop", "MoveItem" },
-                    { "operationID", operationID },
-                    { "filepath", string.Join('|', sourceRename.Select(s => s.Path)) },
-                    { "destpath", string.Join('|', destinationRename) },
-                    { "overwrite", false },
-                    { "HWND", NativeWinApiHelper.CoreWindowHandle.ToInt64() }
-                });
-                result &= (FilesystemResult)(status == AppServiceResponseStatus.Success
-                    && response.Get("Success", defaultJson).GetBoolean());
-                var shellOpResult = JsonSerializer.Deserialize<ShellOperationResult>(response.Get("Result", defaultJson).GetString());
-                moveResult.Items.AddRange(shellOpResult?.Final ?? Enumerable.Empty<ShellOperationItemResult>());
+                var (status, response) = await FileOperationsHelpers.MoveItemAsync(string.Join('|', sourceRename.Select(s => s.Path)), string.Join('|', destinationRename), false, NativeWinApiHelper.CoreWindowHandle.ToInt64(), operationID);
+
+                result &= (FilesystemResult)status;
+                moveResult.Items.AddRange(response?.Final ?? Enumerable.Empty<ShellOperationItemResult>());
             }
             if (sourceReplace.Any())
             {
-                var (status, response) = await connection.SendMessageForResponseAsync(new ValueSet()
-                {
-                    { "Arguments", "FileOperation" },
-                    { "fileop", "MoveItem" },
-                    { "operationID", operationID },
-                    { "filepath", string.Join('|', sourceReplace.Select(s => s.Path)) },
-                    { "destpath", string.Join('|', destinationReplace) },
-                    { "overwrite", true },
-                    { "HWND", NativeWinApiHelper.CoreWindowHandle.ToInt64() }
-                });
-                result &= (FilesystemResult)(status == AppServiceResponseStatus.Success
-                    && response.Get("Success", defaultJson).GetBoolean());
-                var shellOpResult = JsonSerializer.Deserialize<ShellOperationResult>(response.Get("Result", defaultJson).GetString());
-                moveResult.Items.AddRange(shellOpResult?.Final ?? Enumerable.Empty<ShellOperationItemResult>());
+                var (status, response) = await FileOperationsHelpers.MoveItemAsync(string.Join('|', sourceReplace.Select(s => s.Path)), string.Join('|', destinationReplace), true, NativeWinApiHelper.CoreWindowHandle.ToInt64(), operationID);
```

Comment:
Given that the interop between Files.App and Files.FullTrust is no longer needed, we shouldn't serialize paths to a string anymore. Instead we should pass the `string[]` for paths directly.